p2p/protocol/identify/id.go

```diff
@@ -280,21 +314,14 @@ func (ids *IDService) broadcast(proto protocol.ID, payloadWriter func(s network.
 		go func(p peer.ID, conns []network.Conn) {
 			defer wg.Done()
 
-			// if we're in the process of identifying the connection, let's wait.
-			// we don't use ids.IdentifyWait() to avoid unnecessary channel creation.
-		Loop:
+			// Wait till identify completes so we can check the
+			// supported protocols.
 			for _, c := range conns {
-				ids.currmu.RLock()
-				if wait, ok := ids.currid[c]; ok {
-					ids.currmu.RUnlock()
-					select {
-					case <-wait:
-						break Loop
-					case <-ctx.Done():
-						return
-					}
+				select {
+				case <-ids.IdentifyWait(c):
```

Comment:
@Stebalien Why block at all for an incoming `Identify` before sending an outgoing Push/Delta ?